File: Janus/BE/dev/thesis/janus/central/model/RoomCategory.java
package dev.thesis.janus.central.model;

public enum RoomCategory {
    OFFICE("Office", "#FFC107"),
    CONFERENCE("Conference Room", "#8BC34A"),
    LAB("Laboratory", "#03A9F4"),
    LOUNGE("Lounge", "#E91E63"),
    STORAGE("Storage", "#9C27B0"),
    DEFAULT("Default", "#FF0000");

    private final String displayName;
    private final String colorHex;

    RoomCategory(String displayName, String colorHex) {
        this.displayName = displayName;
        this.colorHex = colorHex;
    }

    public String getDisplayName() {
        return displayName;
    }

    public String getColorHex() {
        return colorHex;
    }

    public static RoomCategory fromString(String category) {
        for (RoomCategory roomCategory : RoomCategory.values()) {
            if (roomCategory.displayName.equalsIgnoreCase(category)) {
                return roomCategory;
            }
        }
        return DEFAULT;
    }
}

File: Janus/Kotlin/main/java/com/example/janus/map/RoomCategory.kt
package com.example.janus.map
import android.graphics.Color
enum class RoomCategory(val displayName: String, val color: Int) {
    OFFICE("Office", Color.parseColor("#FFC107")),
    CONFERENCE("Conference Room", Color.parseColor("#8BC34A")),
    LAB("Laboratory", Color.parseColor("#03A9F4")),
    LOUNGE("Lounge", Color.parseColor("#E91E63")),
    STORAGE("Storage", Color.parseColor("#9C27B0")),
    DEFAULT("Default", Color.RED);
    companion object {
        fun fromString(category: String?): RoomCategory {
            return values().find { it.displayName.equals(category, ignoreCase = true) }
                ?: DEFAULT
        }
    }
}
